lib/ecto/adapters/myxql/connection.ex

```diff
@@ -657,6 +657,10 @@ if Code.ensure_loaded?(MyXQL) do
       ["(0 + ", Float.to_string(literal), ?)]
     end
 
+    defp expr(expr, _sources, _query) do
+      raise ArgumentError, "unsupported expression: #{inspect(expr)}"
```

Comment:
```suggestion
    defp expr(expr, _sources, query) do
      error!(query, "unsupported expression: #{inspect(expr)}")
```

Including the query in the error message can be helpful. WDYT?

Reply:
Agreed! I'll do the same for TDS adapter and try to add a test

Reply:
I wouldn't worry about the test because we were supposed to catch those errors in Ecto anyway. I am still trying for the original issue, but this provides a good catch all.